autolab/Autolab
# retry loading file viewer with new header position when render error …

## Description

- Add an exception handler to `submissions_controller.rb` to catch when `format.html { render(:view) }` fails due to it trying to render a non-supported binary file
- Exception handler increments `header_position` (the file it loads) and redirects to the file view to try again until it either runs out of files it can view, which will redirect back to the assessment page, or it loads the first supported file available.
## Motivation and Context
- Fixes #1594
- Solution based on advice from Chaskiel (see issue)
## How Has This Been Tested?



- Submit an archive to an assessment(use this archive: [brokenarchive.tgz](https://github.com/autolab/Autolab/files/9659410/brokenarchive.tgz))
 - Note: this file contains .DS_Store and ._wifitransmitter.py, which aren't displayable and would normally cause an error
- Click on view source
- see that it loads wifitransmitter.py without issue (might take a couple seconds)
## Types of changes


## Checklist:
